src/apps/competitions/tests/test_phase_migration.py

```diff
@@ -0,0 +1,106 @@
+        self.phase2 = PhaseFactory(competition=self.competition, auto_migrate_to_this_phase=True,
+                                   start=five_minutes_ago, index=2, name='Phase2')
+        self.phase3 = PhaseFactory(competition=self.competition, auto_migrate_to_this_phase=False,
+                                   index=3, name='Phase3')
+
+        # Competition 2 Phases
+        self.phase1_competition2 = PhaseFactory(competition=self.competition2, auto_migrate_to_this_phase=False,
+                                                end=twenty_minutes_ago, has_been_migrated=True,
+                                                index=1)
+        self.phase2_competition2 = PhaseFactory(competition=self.competition2, auto_migrate_to_this_phase=True,
+                                                start=five_minutes_ago, index=2)
+
+        # Competition 1 Submissions
+        self.submission = self.make_submission()
+        self.submission2 = self.make_submission(participant=self.competition_participant2)
+
+        # Competition 2 Submissions
+        self.submission1_competition2 = self.make_submission(phase=self.phase1_competition2,
+                                                             participant=self.competition_participant)
+        self.submission2_competition2 = self.make_submission(phase=self.phase1_competition2,
+                                                             participant=self.competition_participant3)
+
+    def make_submission(self, **kwargs):
+        kwargs.setdefault('owner', self.owner)
+        kwargs.setdefault('participant', self.competition_participant)
+        kwargs.setdefault('phase', self.phase1)
+        kwargs.setdefault('status', 'None')
+        sub = SubmissionFactory(**kwargs)
+        SubmissionScoreFactory(submission=sub)
+        return sub
+
+    def test_migrate_submissions(self):
+        assert self.phase2.submissions.count() == 0
+        do_phase_migrations.apply()
+        assert self.phase1.submissions.count() == self.phase2.submissions.count()
+
+    def test_currently_migrating_competitions_dont_migrate(self):
+        self.competition.is_migrating = True
+        self.competition.save()
+        do_phase_migrations.apply()
+        self.assertFalse(self.phase1.submissions.count() == self.phase2.submissions.count())
+
+    def test_competitions_with_scoring_submissions_dont_migrate(self):
+        self.submission3 = self.make_submission(status='Scoring', participant=self.competition_participant3)
+        do_phase_migrations.apply()
+        self.assertFalse(self.phase1.submissions.count() == self.phase2.submissions.count())
+
+    def test_auto_migration_at_new_phase(self):
+        do_phase_migrations.apply()
+        assert self.phase1.submissions.count() == self.phase2.submissions.count()
+
+    def test_submission_ran_after_migration_complete(self):
+        do_phase_migrations.apply()
+        for submission in self.phase2.submissions.all():
```

Comment:
`assert not self.phase2.submissions.filter(status='None').exists()`

Reply:
Changed from loop to filter in latest push